daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2258,6 +2291,37 @@ func (devices *DeviceSet) exportDeviceMetadata(hash string) (*deviceMetadata, er
 	return metadata, nil
 }
 
+func xfsSupported() bool {
+	// Return true only if kernel supports xfs and mkfs.xfs is available
+	exec.Command("modprobe", "xfs").Run()
+
+	f, err := os.Open("/proc/filesystems")
+	if err != nil {
+		return false
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if strings.Contains(s.Text(), "xfs") {
+			if _, err := exec.LookPath("mkfs.xfs"); err == nil {
+				return true
+			}
+		}
+	}
```

Comment:
Ok, so at the end of s.Scan() for loop, I need to make sure scanner did not encounter any errors (except EOF) and if it did, then log a warning. Makes sense. Will do.